api/app/Console/Commands/UpdateApplicationTimestamps.php

```diff
@@ -0,0 +1,158 @@
+
+    private $dry = false;
+
+    private $nullStatuses = [
+        PoolCandidateStatus::NEW_APPLICATION->name,
+        PoolCandidateStatus::APPLICATION_REVIEW->name,
+        PoolCandidateStatus::SCREENED_IN->name,
+        PoolCandidateStatus::UNDER_ASSESSMENT->name,
+    ];
+
+    private $finalStatuses = [
+        PoolCandidateStatus::SCREENED_OUT_APPLICATION->name,
+        PoolCandidateStatus::SCREENED_OUT_ASSESSMENT->name,
+        PoolCandidateStatus::QUALIFIED_AVAILABLE->name,
+        PoolCandidateStatus::QUALIFIED_UNAVAILABLE->name,
+        PoolCandidateStatus::QUALIFIED_WITHDREW->name,
+        PoolCandidateStatus::PLACED_TENTATIVE->name,
+        PoolCandidateStatus::EXPIRED->name,
+    ];
+
+    private $removedStatuses = [
+        PoolCandidateStatus::SCREENED_OUT_NOT_INTERESTED->name,
+        PoolCandidateStatus::SCREENED_OUT_NOT_RESPONSIVE->name,
+        PoolCandidateStatus::REMOVED->name,
+    ];
+
+    private $placedStatuses = [
+        PoolCandidateStatus::PLACED_INDETERMINATE->name,
+        PoolCandidateStatus::PLACED_TERM->name,
+        PoolCandidateStatus::PLACED_CASUAL->name,
+    ];
+
+    /**
+     * Execute the console command.
+     */
+    public function handle()
+    {
+        $this->dry = $this->option('dry');
+
+        $now = now();
+
+        // Null out all timestamps
+        $nullCandidates = PoolCandidate::whereIn('pool_candidate_status', $this->nullStatuses)
+            ->where(function ($query) {
+                $query->whereNotNull('removed_at')
+                    ->orWhereNotNull('placed_at')
+                    ->orWhereNotNull('final_decision_at');
+            });
+
+        $nullCount = $this->update($nullCandidates, [
+            'removed_at' => null,
+            'placed_at' => null,
+            'final_decision_at' => null,
+        ]);
+
+        // Null non final decision fields
+        $final = PoolCandidate::whereIn('pool_candidate_status', $this->finalStatuses)
+            ->where(function ($query) {
+                $query->whereNotNull('removed_at')
+                    ->orWhereNotNull('placed_at');
+            });
+
+        $finalCount = $this->update($final, [
+            'removed_at' => null,
+            'placed_at' => null,
+        ]);
+
+        // Set final decision where null
+        $finalNull = PoolCandidate::whereIn('pool_candidate_status', $this->finalStatuses)
+            ->whereNull('final_decision_at');
+
+        $finalCount += $this->update($finalNull, ['final_decision_at' => $now]);
+
+        // Null non removed at fields
+        $removed = PoolCandidate::whereIn('pool_candidate_status', $this->removedStatuses)
+            ->where(function ($query) {
+                $query->whereNotNull('final_decision_at')
+                    ->orWhereNotNull('placed_at');
+            });
+
+        $removedCount = $this->update($removed, [
+            'final_decision_at' => null,
```

Comment:
Sorry again for the last minute change I sprung on you yesterday.  Please let `final_decision_at` for REMOVED float and not update it.

Reply:
okay, updated in [do not override final decsion for removed status](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12734/commits/313d17f85e20fb3f894e06a046e7e57c79f49f7d)